### PR TITLE
Fix issue with two midrolls in one cue point

### PR DESCRIPTION
**Pull request by VladyslavAnokhin**
### Issue
In case of two and more mid rolls in one cue points we should play only first one.
But nice to have possibility to play that mid roll if user seeks back.

### Fix
To track if user seeks back I've added property which stores time on previous step. If this time bigger than current time we have to play mid roll from cue point which we've already played.
___
Also this PR contains few fixes for telemetry

@VerizonAdPlatforms/video-partner-sdk-developers: Please review.

### CODE REVIEW

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift [hunk 1]
```diff
@@ -32,18 +34,21 @@ final class StartAdProcessingController {
                        isPlayMidrollAllowed: item.content.isPaused == false && item.content.isSeeking == false)
     }
     
+    func startNewSessions(sessionID: UUID) {
+        if self.sessionID != sessionID {
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: `guard`?

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift [hunk 2]
```diff
@@ -19,9 +20,10 @@ final class StartAdProcessingController {
     }
     
     func process(props: Player.Properties) {
+        startNewSessions(sessionID: props.session.playback.id)
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: `startSession`?

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift [hunk 3]
```diff
@@ -32,18 +34,21 @@ final class StartAdProcessingController {
                        isPlayMidrollAllowed: item.content.isPaused == false && item.content.isSeeking == false)
     }
     
+    func startNewSessions(sessionID: UUID) {
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: `private` ?

**Review by AndriiMoskvin — DISMISSED** [review]

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift (hunk 3 above)
Comment (VladyslavAnokhin) [reply sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: I use it it tests

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift (hunk 1 above)
Comment (VladyslavAnokhin) [reply sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: done

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift (hunk 2 above)
Comment (VladyslavAnokhin) [reply sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: done

File: sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift [hunk 4]
```diff
@@ -5,79 +5,120 @@ import PlayerCore
 
 final class StartAdProcessingController {
```
Comment (AndriiMoskvin) [inline sources/advertisements/VRM New Core/Controllers/StartAdProcessingController.swift]: No sense in having this controller - just use preroll and midroll controller with separate logic.

**Review by AndriiMoskvin — DISMISSED** [review]

File: VerizonVideoPartnerSDK.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -51,9 +51,9 @@
...
 		06E5F33821C2CA7E0099D733 /* PlayerCore.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 06E5F31D21C2C7F80099D733 /* PlayerCore.framework */; };
 		06E5F33921C2CA7E0099D733 /* VideoRenderer.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 06E5F31521C2C7EB0099D733 /* VideoRenderer.framework */; };
-		06F8DBC421C2A2D40019021D /* StartAdProcessingController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 06F8DBC121C2A2D40019021D /* StartAdProcessingController.swift */; };
-		06F8DBC521C2A2D40019021D /* StartAdProcessingController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 06F8DBC121C2A2D40019021D /* StartAdProcessingController.swift */; };
-		06F8DBC621C2A2F50019021D /* StartAdProcessingControllerTest.swift in Sources */ = {isa = PBXBuildFile; fileRef = 06F8DBC021C2A2D40019021D /* StartAdProcessingControllerTest.swift */; };
+		06F8DBC421C2A2D40019021D /* VRMPrerollProcessorContorller.swift in Sources */ = {isa = PBXBuildFile; fileRef = 06F8DBC121C2A2D40019021D /* VRMPrerollProcessorContorller.swift */; };
```
Comment (AndriiMoskvin) [inline VerizonVideoPartnerSDK.xcodeproj/project.pbxproj]: Typo here

File: sources/VVPSDK.swift [hunk 6]
```diff
@@ -408,12 +408,12 @@ public struct VVPSDK {
         
         func setupVRMWithNewCore() {
             let maxRedirectCount = player.model.adSettings.maxVASTWrapperRedirectCount
-            let prefetchOffset = player.model.adSettings.prefetchingOffset
             let maxAdSearchTime = player.model.adSettings.maxSearchTime
             let createRequest: (URL) -> (URLRequest) = {
                 .init(url: $0, timeoutInterval: hardTimeout)
             }
-            let adStartProcessing = StartAdProcessingController(prefetchOffset: prefetchOffset, dispatch: dispatcher)
+            let prerollProcessor = VRMPrerollProcessorContorller(dispatch: dispatcher)
```
Comment (AndriiMoskvin) [inline sources/VVPSDK.swift]: typo

**Review by AndriiMoskvin — DISMISSED** [review]

**Review by balavor — DISMISSED** [review]

**Review by balavor — APPROVED** [review]

**Pull request merged**